enumerable_project.rb

```diff
@@ -0,0 +1,387 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+
+# rubocop:disable Metrics/PerceivedComplexity
+
+# rubocop:disable Metrics/ModuleLength
+
+module Enumerable
+  def my_each
+    if self.class == Array
+
+      0.upto(self.length - 1) do |i|
+        yield(self[i])
+      end
+
+    elsif self.class == Hash
+
+      for key in self
+
+        yield(key[0].to_s, key[1])
+
+      end
+
+    end
+  end
+
+  def my_each_with_index
+    if self.class == Array
+
+      0.upto(self.length - 1) do |i|
+        yield(self[i], i)
+      end
+
+    elsif self.class == Hash
+
+      i = 0
+
+      for key in self
+
+        yield([key[0].to_s, key[1]], i)
+
+        i+=1
+
+      end
+
+    end
+  end
+
+  def my_select
+    if self.class == Array
+
+      arr = []
+
+      0.upto(self.length-1) do |i|
+        arr << self[i] if yield(self[i])
+      end
+
+      return arr
+
+    elsif self.class == Hash
+
+      hash = {}
+
+      for key in self
+
+        hash[key[0]] = key[1] if yield(key[0], key[1])
+
+      end
+
+      return hash
+
+    end
+  end
+
+  def my_all?
+    if self.class == Array
+
+      0.upto(self.length-1) do |i|
+        return false unless yield(self[i])
+      end
+
+      return true
+
+    elsif self.class == Hash
+
+      for key in self
+
+        return false unless yield(key[0], key[1])
+
+      end
+
+      return true
+
+    end
+  end
+
+  def my_none?
+    if self.class == Array
+
+      0.upto(self.length-1) do |i|
+        return false if yield(self[i])
+      end
+
+      return true
+
+    elsif self.class == Hash
+
+      for key in self
+
+        return false if yield(key[0], key[1])
+
+      end
+
+      return true
+
+    end
+  end
+
+  def my_count
+    count = 0
+
+    if self.class == Array
+
+      0.upto(self.length-1) do |i|
+        count += 1 if yield(self[i])
+      end
+
+    elsif self.class == Hash
+
+      for key in self
+
+        count += 1 if yield(key[0], key[1])
+
+      end
+
+    end
+
+    return count
+  end
+
+  def my_map(my_proc = nil)
+    arr = []
+
+    if self.class == Array
+
+      0.upto(self.length - 1) do |i|
+        arr << (my_proc ? my_proc.call(i) : yield(self[i]))
+      end
+
+    elsif self.class == Hash
+
+      for key in self
+
+        arr << (my_proc ? my_proc.call(key[0], key[1]) : yield(key[0], key[1]))
+
+      end
+
+    end
+
+    return arr
+  end
+
+  def my_inject(initial = nil)
+    unless initial
+
+      initial = 0 if self.first.class == Integer || self.first.class == Float
+
+      initial = '' if self.first.class == String
+
+      initial = [] if self.first.class == Array
+
+      initial = {} if self.first.class == Hash
+
+    end
+
+    total = initial
+
+    if self.class == Array
+
+      0.upto(self.length-1) do |i|
+        total = yield(total, self[i])
+      end
+
+    elsif self.class == Hash
+
+      for key in self
+
+        total = yield(total, [key[0], key[1]])
+
+      end
+
+    end
+
+    return total
+  end
+end
```

Comment:
- [ ] should have working `my_inject` **(covered by automated tests)** identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))

####  my_inject

- [ ] combines all elements of enum by applying a binary operation, specified by a block
```ruby
(1..5).my_inject(4) { |prod, n| prod * n } == (1..5).inject(4) { |prod, n| prod * n } => 480
```
- Yours is retrieving the wrong result
![image](https://user-images.githubusercontent.com/52209067/78953742-1ccbf380-7a97-11ea-9011-c1cf375ace99.png)

- Real inject
![image](https://user-images.githubusercontent.com/52209067/77698409-2ffaa180-6f76-11ea-91d3-9778aa560411.png)

- [ ] When a symbol is provided as an argument, my_inject should combine the elements of the enum using that symbol as a named method.
```ruby
[2, 5, 6].my_inject(:+) == [2, 5, 6].inject(:+) => 13
```
- Yours is always expecting a block
![image](https://user-images.githubusercontent.com/52209067/78953779-2ead9680-7a97-11ea-9579-8ec53dfe896e.png)

- Real inject
![image](https://user-images.githubusercontent.com/52209067/77698534-6a643e80-6f76-11ea-9d0e-ebb0f02da179.png)